packages/ozone/src/mod-service/status.ts

```diff
@@ -126,14 +126,79 @@ const getSubjectStatusForModerationEvent = ({
   }
 }
 
+const recordEvents = [
+  'tools.ozone.moderation.defs#accountEvent',
+  'tools.ozone.moderation.defs#identityEvent',
+  'tools.ozone.moderation.defs#recordEvent',
+]
+
+const getSubjectStatusForRecordEvent = ({
+  event,
+  currentStatus,
+}: {
+  event: ModerationEventRow
+  currentStatus?: ModerationSubjectStatusRow
+}): Partial<ModerationSubjectStatusRow> => {
+  const timestamp = event.meta?.timestamp
+    ? `${event.meta?.timestamp}`
+    : event.createdAt
+  if (event.action === 'tools.ozone.moderation.defs#recordEvent') {
+    if (event.meta?.op === 'delete') {
+      return {
+        recordStatus: 'deleted',
+        recordDeletedAt: timestamp,
+      }
+    } else if (event.meta?.op === 'update') {
+      return {
+        recordUpdatedAt: timestamp,
+      }
+    }
+    return {}
+  }
+
+  if (event.action === 'tools.ozone.moderation.defs#accountEvent') {
+    const status: Partial<ModerationSubjectStatusRow> = {
+      recordStatus: `${event.meta?.status}`,
+    }
+
+    if (event.meta?.status === 'deleted') {
+      status.recordDeletedAt = timestamp
+    } else {
+      status.recordUpdatedAt = timestamp
+      // When deactivated accounts are re-activated, we receive the event with just the active flag set to true
+      // so we want to make sure that the recordStatus is not set to an outdated value
+      if (currentStatus?.recordStatus !== 'active' && event.meta?.active) {
+        status.recordStatus = 'active'
```

Comment:
default for `recordStatus` is null, may be we should fallback to that? although, doing this is maybe helpful if we want to see which accounts `reactivated`.

Reply:
I'd recommend reflecting the exact hosting status here (where active is either `null` or `'active'`, slight preference for the latter), and then if we want to keep tract of reactivations we do so with a separate field such as `reactivatedAt`.  That allows the status to not be dependent on past statuses, and if we want to know if something is active we don't need to check for multiple different values.

Reply:
yeah the `null` value is basically "we haven't tracked any hosting status for this" so I think safer option is to check for actual `deactivated` value for status.